cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -302,12 +311,38 @@ def __init__(
                 s3_key=f"{bucket.artifact_directory}/custom_resources/artifacts.zip",
             ),
             handler=f"{handler_func}.handler",
-            memory_size=128,
+            memory_size=memory_size,
             role=execution_role,
             runtime="python3.8",
             timeout=timeout,
+            environment=awslambda.CfnFunction.EnvironmentProperty(variables=environment),
         )
 
+        if schedule_expression is not None:
+            self.lambda_rule = events.CfnRule(
```

Comment:
is there a reason `CfnRule` has to be used instead of the `Rule` construct? probably there is but i'm admittedly not very familiar with this part of the 3.0 code.

Reply:
Generally speaking, Cfn* constructs are preferable over higher level constructs when you want more control over the final template. On the other hand, higher level constructs are more comfortable because they exposes handy signatures and take care of some underlying resources. I think that, when it is possible Cfn* constructs are preferable aklso because, at the end, higher level constructs are translated to low level ones.